x/claim/keeper/claim_test.go

```diff
@@ -667,3 +669,128 @@ func (suite *KeeperTestSuite) TestGetClaimMetadata() {
+		suite.Require().NoError(err)
+		err = suite.app.BankKeeper.SendCoins(suite.ctx, addr, distributors[airdropId], sdk.NewCoins(sdk.NewInt64Coin(sdk.DefaultBondDenom, initialBal)))
+		suite.Require().NoError(err)
+	}
+
+	claimRecords := []types.ClaimRecord{
+		{
+			Address:           addr1.String(),
+			Weight:            weight, // 50%
+			ActionCompleted:   []bool{false, false, false},
+			AirdropIdentifier: airdropId,
+		},
+	}
+
+	err = suite.app.ClaimKeeper.SetClaimRecordsWithWeights(suite.ctx, claimRecords)
+	suite.Require().NoError(err)
+
+	return airdropId, EVMOS_ADDR, addr1.String(), STRIDE_ADDR
+}
+
+func (suite *KeeperTestSuite) TestUpdateAirdropAddress() {
+	airdropId, evmosAddress, evmosParsedAddress, strideAddress := suite.SetupUpdateAirdropAddressChangeTests()
+	evmosParsedAccAddress, err := sdk.AccAddressFromBech32(evmosParsedAddress)
+	suite.Require().NoError(err)
+	strideAccAddress, err := sdk.AccAddressFromBech32(strideAddress)
+	suite.Require().NoError(err)
+
+	airdropClaimCoins := sdk.NewCoins(sdk.NewInt64Coin(sdk.DefaultBondDenom, 100_000_000))
+
+	// verify that the Evmos address changed
+	suite.Require().NotEqual(evmosAddress, evmosParsedAddress, "evmos address should have changed")
+	// verify new Evmos address starts with "stride"
+	suite.Require().True(strings.HasPrefix(evmosParsedAddress, "stride"), "evmos address should start with stride")
+
+	coins, err := suite.app.ClaimKeeper.GetUserTotalClaimable(suite.ctx, evmosParsedAccAddress, airdropId, false)
+	suite.Require().NoError(err)
+	suite.Require().Equal(coins.String(), airdropClaimCoins.String())
+
+	// verify that we can't claim the airdrop with the original stride address
+	coins, err = suite.app.ClaimKeeper.GetUserTotalClaimable(suite.ctx, strideAccAddress, airdropId, false)
+	suite.Require().NoError(err)
+	emptyCoins := sdk.NewCoins()
+	suite.Require().Equal(coins, emptyCoins, "stride address should claim 0 coins before update", strideAccAddress)
+	claims, err := suite.app.ClaimKeeper.GetClaimStatus(suite.ctx, strideAccAddress)
+	suite.Require().NoError(err)
+	suite.Require().Empty(claims, "stride address should have 0 claim records before update")
+
+	// verify that evmos can claim the address
+	coins, err = suite.app.ClaimKeeper.GetUserTotalClaimable(suite.ctx, evmosParsedAccAddress, airdropId, false)
+	suite.Require().NoError(err)
+	suite.Require().Equal(coins, sdk.NewCoins(sdk.NewCoin("stake", sdk.NewIntFromUint64(100_000_000))), "parsed evmos address should be allowed to claim")
+	claims, err = suite.app.ClaimKeeper.GetClaimStatus(suite.ctx, evmosParsedAccAddress)
+	suite.Require().NoError(err)
+	properClaims := []types.ClaimStatus{{AirdropIdentifier: airdropId, Claimed: false}}
+	suite.Require().Equal(claims, properClaims, "evmos address should have 1 claim record before update")
+
+	// update the address
+	success := suite.app.ClaimKeeper.UpdateAirdropAddress(suite.ctx, evmosParsedAddress, strideAddress, airdropId)
```

Comment:
nit: would be slightly cleaner if UpdateAirdropAddress returned an error

Reply:
Agreed, changed!